terraspace/app/stacks/dns/provider.tf

```diff
@@ -0,0 +1,2 @@
+provider "tls" {}
+provider cloudflare {}
```

Comment:
Do you need these block in conjunction with the declarations in `required_providers` ?

Reply:
> Do you need these block in conjunction with the declarations in required_providers ?

Yes

Reply:
> Additionally, I don't see anything in this PR that authorizes GHA to communicate with Cloudflare API.

I've added the required environment variables to the workflow and secrets.

Reply:
This request about updating the workflow with credentials was a good reminder of these technical debts

- https://www.notion.so/Elastic-IPFS-log-868adc5d3a8f4094bf5cd0eb26679c15?p=b2c874b5b037410698a32ccd10a04e0e&pm=s
- https://www.notion.so/Elastic-IPFS-log-868adc5d3a8f4094bf5cd0eb26679c15?p=bdc2847e1e1341898602df5df5c755be&pm=s

Reply:
I've tried to simply run the workflow on the self runner but that came with a bunch of other necessary improvements (Rolled back)

Reply:
I temporarily removed the failed stack to see the plan for DNS stack:
https://github.com/elastic-ipfs/infrastructure/runs/7793501635?check_suite_focus=true

Reply:
> I temporarily removed the failed stack to see the plan for DNS stack: https://github.com/elastic-ipfs/infrastructure/runs/7793501635?check_suite_focus=true

Remember, you should be able to create a PR comment such as "terraspace plan dns" to avoid having to remove the stack entirely.